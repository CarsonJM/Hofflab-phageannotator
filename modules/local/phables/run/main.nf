process PHABLES_RUN {
    tag "${meta.id}"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://carsonjm/phables:1.4.0':
        'docker://carsonjm/phables:1.4.0' }"

    input:
    tuple val(meta) , path(graph)
    tuple val(meta2), path(fastq)
    path phables_db

    output:
    tuple val(meta), path("${prefix}_phables.fasta.gz") , emit: fasta
    tuple val(meta), path("${prefix}_phables.log")      , emit: log
    path "versions.yml"                                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    bash /usr/local/bin/_entrypoint.sh
    mkdir reads
    cp ${fastq[0]} reads/
    cp ${fastq[1]} reads/
    mkdir phables.out

    phables \\
        run \\
        --configfile ${projectDir}/assets/phables/config.yml \\
        --input ${graph} \\
        --reads reads \\
        --threads ${task.cpus} \\
        --prefix ${prefix} \\
        ${args}

    mv phables_out/phables/resolved_paths.fasta ./${prefix}_phables.fasta
    gzip ./${prefix}_phables.fasta
    mv phables_out/phables.log ./${prefix}_phables.log
    rm -rf phables_out/

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        phables: \$(echo \$(phables -v 2>&1) | sed -n 's/phables, version //' ))
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_phables.log
    echo "" | gzip > ${prefix}_phables.fasta.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        phables: \$(echo \$(phables -v 2>&1) | sed -n 's/phables, version //' ))
    END_VERSIONS
    """
}
