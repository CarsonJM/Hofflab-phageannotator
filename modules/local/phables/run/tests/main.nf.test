nextflow_process {

    name "Test Process: PHABLES_RUN"
    script "../main.nf"
    process "PHABLES_RUN"

    setup {
        run("PHABLES_INSTALL") {
            script "../../install/main.nf"
            }
        }


    test("fastq.gz + gfa.gz") {

        when {
            process {
                """
                input[0] = [
                    [ id: 'test1' ],
                    [
                        file("${projectDir}/assets/test-datasets/gfa/test1.assembly.gfa.gz", checkIfExists: true),
                    ]
                ]
                input[1] = [
                    [ id: 'test1' ],
                    [
                        file("${projectDir}/assets/test-datasets/fastqs/test1_1.fastq.gz", checkIfExists: true),
                        file("${projectDir}/assets/test-datasets/fastqs/test1_2.fastq.gz", checkIfExists: true)
                    ]
                ]
                input[2] = PHABLES_INSTALL.out.phables_db
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out.fasta).match() },
                { assert new File(process.out.log[0][1]).exists() },
            )
        }
    }

    test("fastq.gz - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id: 'test1' ],
                    [
                        file("${projectDir}/assets/test-datasets/gfa/test1.assembly.gfa.gz", checkIfExists: true),
                    ]
                ]
                input[1] = [
                    [ id: 'test1' ],
                    [
                        file("${projectDir}/assets/test-datasets/fastqs/test1_1.fastq.gz", checkIfExists: true),
                        file("${projectDir}/assets/test-datasets/fastqs/test1_2.fastq.gz", checkIfExists: true)
                    ]
                ]
                input[2] = PHABLES_INSTALL.out.phables_db
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() },
            )
        }
    }
}
