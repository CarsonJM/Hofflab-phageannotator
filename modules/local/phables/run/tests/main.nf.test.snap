{
    "fastq.gz + gfa.gz": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-12T13:36:20.629861949"
    }
}